app/services/catalog_record_id.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+class CatalogRecordId
+  # NOTE: This block is a junk drawer of simple class methods meant to help
+  #       switch between Symphony and Folio catalog implementations, and can be
+  #       dumped or in-lined once we're done with Symphony for good.
+  class << self
+    def other_links(model)
+      new(model).other_links.map(&:catalogRecordId)
+    end
+
+    def valid?(catalog_record_ids)
+      catalog_record_ids.all? { |id| id.match?(Regexp.new(pattern_string)) }
+    end
+
+    def label
+      return "Folio ID" if Settings.enabled_features.folio
+
+      "Catkey"
+    end
+
+    def manage_label
+      return "Manage Folio ID" if Settings.enabled_features.folio
+
+      "Manage catkey"
+    end
+
+    def index_field
+      return SolrDocument::FIELD_FOLIO_INSTANCE_HRID if Settings.enabled_features.folio
+
+      SolrDocument::FIELD_CATKEY_ID
+    end
+
+    def pattern_string
+      return "\\A(L|a|in)[0-9]+\\z" if Settings.enabled_features.folio
+
+      "\\A[0-9]+(:[0-9]+)*\\z"
+    end
+
+    def html_pattern_string
+      return "^(L|a|in)[0-9]+$" if Settings.enabled_features.folio
+
+      "^[0-9]+(:[0-9]+)*$"
+    end
+
+    def index_field_prefix
+      return "folio" if Settings.enabled_features.folio
+
+      "catkey"
+    end
+
+    def value
+      return "folio" if Settings.enabled_features.folio
+
+      "symphony"
+    end
+
+    def previous_value
+      return "previous folio" if Settings.enabled_features.folio
+
+      "previous symphony"
+    end
+  end
+
+  def self.links(model)
+    new(model).links
+  end
+
+  def self.serialize(model, catalog_record_ids, refresh: true)
+    new(model).serialize(catalog_record_ids, refresh:)
+  end
+
+  def self.link_refresh(model)
+    new(model).link_refresh
+  end
+
+  def initialize(model)
+    @model = model
+  end
+
+  # If there was already a catalog record ID in the record, store that in the "previous" spot (assuming there is no change)
+  # @param [Array<String>] new_catalog_record_ids a list of catalog record IDs
+  # @param [boolean] refresh first catalog record ID
+  # @return [Array<Hash>] a list of catalog links
+  def serialize(new_catalog_record_ids, refresh: true)
+    removed_links = links - new_catalog_record_ids
+    links = (previous_links + removed_links).map { |record_id| {catalog: self.class.previous_value, catalogRecordId: record_id, refresh: false} }.uniq
+    links +
+      other_links.map(&:to_h) +
+      new_catalog_record_ids.map.with_index { |record_id, index| {catalog: self.class.value, catalogRecordId: record_id, refresh: refresh && index.zero?} }
+  end
+
+  def links
+    find(self.class.value)
+  end
+
+  def other_links
+    find_not(self.class.value)
+  end
+
+  def link_refresh
```

Comment:
possibly naive question:  so, is there no need to refresh from a link to the "other" catalog?  i can see how that makes sense, just wondering since `CatalogRecordIdForm` has a comment saying: "at most one catalog record ID per catalog (Symphony/Folio) can be set to refresh == true"

Reply:
As far as I know, yes. We'll refresh only from the enabled catalog.

Reply:
good question for A & A.

Reply:
I believe the "at most one of each can have refresh=true" is related to the validation rules for refresh links. In a Symphony-only world, only one link could be refresh=true. But in a world where both catalogs are represented, where the data must remain in sync, the rule had to be changed so both Symphony and Folio could have one refresh=true. Otherwise, you wouldn't be able to refresh when switching between systems.